tests/e2e/configurer/upgrade.go

```diff
@@ -147,10 +147,31 @@ func (uc *UpgradeConfigurer) RunSetup() error {
 }
 
 func (uc *UpgradeConfigurer) RunUpgrade() error {
+	var err error
 	if uc.forkHeight > 0 {
-		return uc.runForkUpgrade()
+		err = uc.runForkUpgrade()
+	} else {
+		err = uc.runProposalUpgrade()
 	}
-	return uc.runProposalUpgrade()
+	if err != nil {
+		return err
+	}
+
+	// Check if the nodes are running
+	for chainIndex, chainConfig := range uc.chainConfigs {
+		chain := uc.baseConfigurer.GetChainConfig(chainIndex)
+		for validatorIdx := range chainConfig.NodeConfigs {
+			node := chain.NodeConfigs[validatorIdx]
+			// Check node status
+			_, err = node.Status()
```

Comment:
just out of curiousity, do we not need to check on the status returned but only do err checking to determine if the node is running / or not?

Reply:
If we get a correct response from `osmosisd status` and it's in the desired format (can be parsed), we can conclude that the node is running.

I don't think any of the result fields are useful for our purposes since we just want to make sure that a node is running

Reply:
From the last check, seem that we cant unmarshal the `Status` response to the original struct cause it's already converted to string obj.
So I think just check the cmd response & no return err is enough. cc @RusAkh

Reply:
And use `any` type to unmarshal